src/reaction.rs

```diff
@@ -0,0 +1,471 @@
+//! # Reactions.
+//!
+//! Reactions are short messages consisting of emojis sent in reply to
+//! messages. Unlike normal messages which are added to the end of the chat,
+//! reactions are supposed to be displayed near the original messages.
+//!
+//! RFC 9078 specifies how reactions are transmitted in MIME messages.
+//!
+//! Reaction update semantics is not well-defined in RFC 9078, so
+//! Delta Chat uses the same semantics as in
+//! [XEP-0444](https://xmpp.org/extensions/xep-0444.html) section
+//! "3.2 Updating reactions to a message". Received reactions override
+//! all previously received reactions from the same user and it is
+//! possible to remove all reactions by sending an empty string as a reaction,
+//! even though RFC 9078 requires at least one emoji to be sent.
```

Comment:
maybe we should also document this in the chat mail specification (./spec.md), because it differs from the normal specification?